crates/handler/src/pre_execution.rs

```diff
@@ -106,6 +106,27 @@ pub fn validate_account_nonce_and_code(
     Ok(())
 }
 
+/// Make changes to the caller account.
+///
+/// Returns the old balance.
+#[inline]
+pub fn caller_touch_and_change(
+    caller_account: &mut Account,
```

Comment:
should this be moved into `impl Account`?

Reply:
Yes